contracts/eden/src/accounts.cpp

```diff
@@ -0,0 +1,47 @@
+#include <accounts.hpp>
+
+namespace eden
+{
+   std::optional<account> accounts::get_account(eosio::name owner)
+   {
+      auto record = account_tb.find(owner.value);
+      if (record != account_tb.end())
+         return *record;
+      return std::nullopt;
+   }
+
+   void accounts::add_balance(eosio::name owner, const eosio::asset& quantity)
+   {
+      auto record = account_tb.find(owner.value);
+      if (record == account_tb.end())
+      {
+         // TODO: create another global
+         eosio::check(quantity >= globals.get().minimum_donation,
+                      "insufficient deposit to open an account");
```

Comment:
I think we don't need to have a minimum deposit amount... eg: the user can deposit 5 and 5, and then complete his induction...

Reply:
Since the contract pays for RAM on the row, we need some sort of mitigation against dust. This protection is defeatable by depositing 10.0000 and withdrawing 9.9999, but at least it protects against the common message deposit spam.